src/getchar.c

```diff
@@ -2175,6 +2189,8 @@ do_key_input_pre(int c)
     else
 	buf[(*mb_char2bytes)(c, buf)] = NUL;
 
+    typedchars[typedchars_pos] = NUL;
```

Comment:
You need to call `vim_unescape_csi()` on `typedchars[]` before calling the autocommand (like how that's done when finishing recording a register), otherwise chars like "…" will have K_SPECIAL/CSI escaped.

Reply:
```suggestion
    typedchars[typedchars_pos] = NUL;
    vim_unescape_csi(typedchars);
```